src/Runner.Worker/FileCommandManager.cs

```diff
@@ -182,35 +185,69 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
 
-                using (var streamReader = new StreamReader(filePath))
-                using (var streamWriter = new StreamWriter(scrubbedFilePath))
+                    Trace.Info($"Queueing results file ({filePath}) for attachment upload ({attachmentName})");
+                    // Attachments must be added to the parent context (job), not the current context (step)
+                    context.Root.QueueResultsFile(ChecksAttachmentType.ResultsStepSummary, attachmentName, scrubbedFilePath, stepId);
+                }
+                else 
                 {
-                    string line;
-                    while ((line = streamReader.ReadLine()) != null)
+                    Trace.Info("Using Attachment for Job Summary upload");
+                    if (fileSize > AttachmentSizeLimit)
                     {
-                        var maskedLine = HostContext.SecretMasker.MaskSecrets(line);
-                        streamWriter.WriteLine(maskedLine);
+                        context.Error(String.Format(Constants.Runner.UnsupportedSummarySize, AttachmentSizeLimit / 1024, fileSize / 1024));
+                        Trace.Info($"Step Summary file ({filePath}) is too large ({fileSize} bytes); skipping attachment upload");
+
+                        return;
+                    }
+
+                    Trace.Verbose($"Step Summary file exists: {filePath} and has a file size of {fileSize} bytes");
+
+                    using (var streamReader = new StreamReader(filePath))
+                    using (var streamWriter = new StreamWriter(scrubbedFilePath))
+                    {
+                        string line;
+                        while ((line = streamReader.ReadLine()) != null)
+                        {
+                            var maskedLine = HostContext.SecretMasker.MaskSecrets(line);
+                            streamWriter.WriteLine(maskedLine);
+                        }
                     }
-                }
 
-                var attachmentName = !context.IsEmbedded 
-                    ? context.Id.ToString() 
-                    : context.EmbeddedId.ToString();
+                    var attachmentName = !context.IsEmbedded 
+                        ? context.Id.ToString() 
+                        : context.EmbeddedId.ToString();
 
-                Trace.Info($"Queueing file ({filePath}) for attachment upload ({attachmentName})");
-                // Attachments must be added to the parent context (job), not the current context (step)
-                context.Root.QueueAttachFile(ChecksAttachmentType.StepSummary, attachmentName, scrubbedFilePath);
+                    Trace.Info($"Queueing file ({filePath}) for attachment upload ({attachmentName})");
+                    // Attachments must be added to the parent context (job), not the current context (step)
+                    context.Root.QueueAttachFile(ChecksAttachmentType.StepSummary, attachmentName, scrubbedFilePath);
```

Comment:
would it make sense to only branch on this statement?
```C#
If (useResultService)
{
    context.QueueResultFile()
}
else
{
    context.QueueAttachFile()
}
```

Reply:
Yeah that is a great callout